src/objects/job.py

```diff
@@ -96,11 +96,15 @@ def __init__(
             logs_dir=self.logs_dir,
             junit_dir=self.junit_dir,
         )
-        # self.has_test_failures = self._check_has_test_failures(failures=self.failures)
-        self.has_test_failures = any(failure.failure_type == "test_failure" for failure in self.failures or [])
-        # self.has_pod_failures = self._check_has_pod_failures(failures=self.failures)
-        self.has_pod_failures = any(failure.failure_type == "pod_failure" for failure in self.failures or [])
-        self.has_failures = True if self.failures else False
+
+        self.has_pod_failures = self.has_test_failures = False
+        for failure in self.failures:
+            if failure.failure_type == "test_failure":
+                self.has_test_failures = True
```

Comment:
```suggestion
            if self.has_test_failures := (failure.failure_type == "test_failure"):
                 break
```

Reply:
suggetion

Reply:
@rnetser I tried this last week and was unable to use it because you cannot use the walrus operator with any `self.*` variable on the left side.

[This is the best explanation](https://stackoverflow.com/questions/64055314/why-cant-pythons-walrus-operator-be-used-to-set-instance-attributes) for why that isn't allowed per PEP 572.